t/t2260-job-list.t

```diff
@@ -2213,21 +2213,16 @@ test_expect_success 'job-list returns expected jobspec changes after reload' '
 	flux job list -s inactive | grep $(cat update1.id) | jq -e ".duration == 1000.0"
 '
 
+#
+# Note: Before job-list reload, a job was submitted to defualtqueue but was
+# immediately updated to updatequeue. After reload, no jobs are in updatequeue
+# and job-list currently doesn't update queue stats during KVS reload.
+# Therefore, no stats entry for defaultqueue will appear in `flux job stats`.
+#
```

Comment:
typo "defaultqueue". '

So queue stats actually are updated.  The issue is b/c the lone job has been updated from "defaultqueue" to "updatequeue", so "defaultqueue" has no job / stats in it after reload.

So perhaps something like

After reload, job-list will not have ever seen any jobs submitted to "defaultqueue", therefore the stats object is empty.

It is an open question if this is a bug.  Should job-list read the config and initialize stats to 0 for anything?  I'll open an issue on that.

Reply:
After reload, queue stats are specifically not updated as the eventlog is replayed. I'll fix the typo but I think the description is correct for now.

Reply:
The queue stats have to be updated, otherwise the second half of this test would be all zeros.

```
test_expect_success 'job stats in each queue correct after reload' '
        defaultq=`flux job stats | jq ".queues[] | select( .name == \"defaultqueue\" )"` &&
        updateq=`flux job stats | jq ".queues[] | select( .name == \"updatequeue\" )"` &&
        echo $defaultq | jq -e ".job_states.depend == 0" &&
        echo $defaultq | jq -e ".job_states.priority == 0" &&
        echo $defaultq | jq -e ".job_states.sched == 0" &&
        echo $defaultq | jq -e ".job_states.run == 0" &&
        echo $defaultq | jq -e ".job_states.cleanup == 0" &&
        echo $defaultq | jq -e ".job_states.inactive == 0" &&
        echo $defaultq | jq -e ".job_states.total == 0" &&
        echo $defaultq | jq -e ".successful == 0" &&
        echo $defaultq | jq -e ".failed == 0" &&
        echo $defaultq | jq -e ".canceled == 0" &&
        echo $defaultq | jq -e ".timeout == 0" &&
        echo $defaultq | jq -e ".inactive_purged == 0" &&
        echo $updateq | jq -e ".job_states.depend == 0" &&
        echo $updateq | jq -e ".job_states.priority == 0" &&
        echo $updateq | jq -e ".job_states.sched == 0" &&
        echo $updateq | jq -e ".job_states.run == 0" &&
        echo $updateq | jq -e ".job_states.cleanup == 0" &&
        echo $updateq | jq -e ".job_states.inactive == 1" &&
        echo $updateq | jq -e ".job_states.total == 1" &&
        echo $updateq | jq -e ".successful == 1" &&
        echo $updateq | jq -e ".failed == 0" &&
        echo $updateq | jq -e ".canceled == 0" &&
        echo $updateq | jq -e ".timeout == 0" &&
        echo $updateq | jq -e ".inactive_purged == 0"
'
```

Reply:
> The queue stats have to be updated, otherwise the second half of this test would be all zeros.

Yes, but they are update _after_ KVS reload, not _during_ as noted in the comment.

Reply:
It just occurred to me why this confused me.

> The point of the comment is that the stats are not updated for each eventlog event during replayed, so the original submit queue is lost.

I believe the queue is determined via the jobspec.  I don't think it's in anything in the eventlog.  And the `jobspec-update` event doesn't store the old value.

Reply:
Yeah, that is a fair point. I didn't realize that job-list loads the jobspec _after_ replaying the eventlog, and caches updates until it parses it. If job-list replayed the eventlog in the same way it does during non-restart (parse jobspec, _then_ replay eventlog on top of it, meanwhile updating stats, then this problem wouldn't be an issue)

I'll just update the comment to your suggestion, since it doesn't really matter, though I'll note that job-list *has* seen a job submitted to defaultqueue since that is what is in the job's jobspec. It just doesn't update stats until _after_ it collects and applies all jobspec-update events.

Reply:
> I didn't realize that job-list loads the jobspec after replaying the eventlog

Ahhh I can see that being confusing.  jobspec & eventlog could be done in a different order.  But I think it was done in this order for `R` (gotta determine via job state if it should exist yet).  And jobspec is lumped together with the KVS lookup for R.

Reply:
Ok I updated the comment to your suggested wording @chu11